lib/uplink/apikey.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"time"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// APIKey represents an access credential to certain resources
+type APIKey struct {
+	key string
+}
+
+// Serialize serializes the API Key to a string
+func (a APIKey) Serialize() string {
+	return a.key
+}
+
+// ParseAPIKey parses an API Key
+func ParseAPIKey(val string) (APIKey, error) {
+	return APIKey{key: val}, nil
+}
+
+// Share returns a new API Key with the given caveat restrictions added
+func (a APIKey) Share(caveats ...Caveat) APIKey {
```

Comment:
I suggest removing anything related to sharing from the API for the Vanguard release. We have a separate milestone for Access Management, don't we? We can introduce the respective API with that release.

Reply:
k fair point, just need to revert 0dc098c then

Reply:
i'll do that